### PR TITLE
initial support for PPC64LE

### LINKED ISSUE #317: aarch64 and ppc64 support?
Hi @eliben !
I wrote [a utility to help debug](https://github.com/ClangBuiltLinux/frame-larger-than) `-Wframe-larger-than=` warnings (which are pretty unhelpful in telling you what particular variables may be causing this, especially when many child calls have been inlined.  Apparently all of that info about local variables in captured in the debug info).

We used it for debugging a warning for a ppc64 Linux kernel build, but it seemed the object file failed to parse.  We [observed](https://groups.google.com/g/clang-built-linux/c/GjVJqQMvjvg/m/45gaOrW0AAAJ) the thrown ElfError: `Unsupported relocation type: 1`.

I haven't dug into reproducing yet, but I wanted to check first if pyelftools supports ppc64? I'm not sure if ppc64 uses a different target triple for endianess or not.  I also haven't dumped the relocation kinds, which may be helpful for this report.

### PR DESCRIPTION
**Pull request by nickdesaulniers**
See also:
https://openpowerfoundation.org/wp-content/uploads/2016/03/ABI64BitOpenPOWERv1.1_16July2015_pub4.pdf
3.4.1 Symbol Values
3.5.3 Relocation Types Table

Fixes #317
Signed-off-by: Nick Desaulniers <ndesaulniers@google.com>

### CODE REVIEW

**Comment by nickdesaulniers** [comment]
cc @trevorwhitaker

File: elftools/elf/enums.py [hunk 1]
```diff
@@ -1146,3 +1151,130 @@
...
+    R_PPC64_SECTOFF_DS=61,
+    R_PPC64_SECTOFF_LO_DS=62,
+    R_PPC64_TOC16_DS=63,
+    R_PPC64_TOC16_LO_DS=64,
+    R_PPC64_PLTGOT16_DS=65,
+    R_PPC64_PLTGOT16_LO_DS=66,
+    R_PPC64_TLS=67,
+    R_PPC64_DTPMOD64=68,
+    R_PPC64_TPREL16=69,
+    R_PPC64_TPREL16_LO=60,
```
[the remark this answers is not in the document]
Comment (nickdesaulniers) [reply elftools/elf/enums.py:1225]: oh, interesting. No, I copied these from v1 of the ABI, hit a failure for larger relocation numbers, recalled there was a v2 of the ABI, and added those from the linked doc above.  So either this was wrong in the v1 docs, or perhaps is near the boundary from what was added in the v2 docs.

Nice catch though, I will review the v2 doc and fix this.

**Comment by eliben** [comment]
@trevorwhitaker LGTY?

@nickdesaulniers could you find someone else to help review this?

**Review by trevorwhitaker — APPROVED** [review]

**Review by stephenhines — APPROVED** [review]
Looks pretty straightforward. I also verified all the hyperlinks. :)

File: elftools/elf/descriptions.py [hunk 2]
```diff
@@ -358,6 +373,7 @@ def describe_attr_tag_arm(tag, val, extra):
     SHT_GNU_verneed='VERNEED',
     SHT_GNU_versym='VERSYM',
     SHT_GNU_LIBLIST='GNU_LIBLIST',
+    # SHT_LLVM_ADDRSIG='LLVM_ADDRSIG',
```
Comment (eliben) [inline elftools/elf/descriptions.py:376]: Is this needed?

File: elftools/elf/enums.py [hunk 3]
```diff
@@ -293,6 +293,7 @@
     SHT_SYMTAB_SHNDX=18,
     SHT_NUM=19,
     SHT_LOOS=0x60000000,
+    # SHT_LLVM_ADDRSIG=0x6fff4c03,
```
Comment (eliben) [inline elftools/elf/enums.py:296]: Ditto - is this needed?

File: elftools/elf/descriptions.py (hunk 2 above)
Comment (nickdesaulniers) [reply elftools/elf/descriptions.py]: I probably wont have time to add LLVM ADDR SIG support; I've added a commit on top removing them.  Feel free to squash when merging; if it's kept separate the values will be in git history at least.

**Review by eliben — APPROVED** [review]

**Pull request merged**